### PR TITLE
Refactoring towards a pbstream -> PGM/YAML tool.

### PR DESCRIPTION
**Pull request by SirVer**

### CODE REVIEW

**Comment by SirVer** [comment]
@ensonic @jihoonl Stefan, this is the binary you asked for. Jihoon, this is the binary that I did not let you check in the last time around :P.

**Comment by SirVer** [comment]
Pulled out the refactorings only. The tool will come tomorrow in a followup.

File: cartographer_ros/cartographer_ros/submap.cc [hunk 1]
```diff
@@ -24,6 +24,51 @@
...
+    for (int j = 0; j < width; ++j) {
+      intensity->push_back(cells[(i * width + j) * 2]);
+      alpha->push_back(cells[(i * width + j) * 2 + 1]);
+    }
+  }
+}
+
+::cartographer::io::UniqueCairoSurfacePtr DrawTexture(
+    const std::vector<char>& intensity, const std::vector<char>& alpha,
+    const int width, const int height, std::vector<uint32_t>* cairo_data) {
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/submap.cc]: `std::vector<uint32_t>* const cairo_data`?

File: cartographer_ros/cartographer_ros/submap.cc [hunk 2]
```diff
@@ -37,21 +82,11 @@ std::unique_ptr<SubmapTextures> FetchSubmapTextures(
...
-    intensity.reserve(num_pixels);
     std::vector<char> alpha;
-    alpha.reserve(num_pixels);
-    for (int i = 0; i < texture.height; ++i) {
-      for (int j = 0; j < texture.width; ++j) {
-        intensity.push_back(cells[(i * texture.width + j) * 2]);
-        alpha.push_back(cells[(i * texture.width + j) * 2 + 1]);
-      }
-    }
+    std::string compressed_cells(texture.cells.begin(), texture.cells.end());
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/submap.cc]: `const std::string`?

File: cartographer_ros/cartographer_ros/submap.cc [hunk 3]
```diff
@@ -24,6 +24,51 @@
...
 namespace cartographer_ros {
 
+void UnpackTextureData(const std::string& compressed_cells, int width,
+                       int height, std::vector<char>* intensity,
+                       std::vector<char>* alpha) {
+  std::string cells;
+  ::cartographer::common::FastGunzipString(compressed_cells, &cells);
+  const int num_pixels = width * height;
+  CHECK_EQ(cells.size(), 2 * num_pixels);
+  intensity->reserve(num_pixels);
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/submap.cc]: Clear `intensity` and `alpha`? Or even better return in a struct?

**Comment by SirVer** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user SirVer. Merge queue now has a length of 2.

**Comment by jihoonl** [comment]
:heart_eyes:

**Pull request merged**